crates/subspace-networking/examples/get-pieces-complex.rs

```diff
@@ -83,12 +86,20 @@ async fn main() {
         if i == EXPECTED_NODE_INDEX {
             expected_node_id = node.id();
         }
+        nodes.push(node);
     }
 
+    let db_path = std::env::temp_dir().join("database2").into_boxed_path();
```

Comment:
Please don't, create a temporary directory with `tempfile` crate like we do in other places. This code will leave `database2` in `/tmp` (which on many systems is tmpfs) forever. This isn't nice.

Reply:
It's a temporary directory but it's lifetime goes beyond the process lifetime. We need multiples runs to test required functionality (save and load). I tried `tempfile::tempdir()` and it seems to remove the directory on the process exit which is a good thing but not for this use-case.

Reply:
Hm... I usually don't like when apps silently create random databases. Why do we need to persist it here? Also name `database2` can be quite literally anything when I'm looking at it in my `/tmp`.

Reply:
This app is in example folder :) You don't accidently run examples for libs. But I can rename the temp database name to address its temporary status and actual meaning.

Reply:
Yes, please. And print a message about that in application logs.